servers.controllers.js

```diff
@@ -44,7 +46,16 @@ exports.getByArticleId = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  getAllArticles()
+  const { topic } = req.query;
+  if (!topic) {
+    return getAllArticles().then((article) => {
```

Comment:
getAllArticles is a bit of similar name to getARticles.  Maybe rename this model to selectArticles or fetchArticles